### ISSUE TITLE
Commit new solution files for Data Structures

### ISSUE THREAD

**Issue by Kvaibhav01** [Kvaibhav01/HackerRank-in-Java#12]
Visit [HackerRank's Data Structures website](https://www.hackerrank.com/domains/data-structures), visit the subdomains and commit new problem solutions at the [repo's Data Structures folder](https://github.com/Kvaibhav01/HackerRank-Solutions/tree/master/Data%20Structures).

### Rules:
1. Solutions must be **correct** (obviously), I will review them before merging.
2. The solution's directory path must be **according to the subdomain** HackerRank provides for Data Structures. Example: _/Data Structures/Subdomain name/Solution.java_ or _/Data Structures/Subdomain name/RelevantFileName.java_
3. Code can be done in any preferred language but **Java and JavaScript are recommended**. Someone who provides the solution in Kotlin/PHP (or any other language except recommended ones) will be considered later.
4. Code should be **formatted properly**. Unformatted code will be disregarded.
5. The solution you submit should be of **your own**.

**Comment by rajvijen** [comment]
**Will C++ and Python accepted or not?**

**Comment by nsalazar0429** [comment]
When can I start working in this?

**Comment by aayushbhan** [comment]
Hey, can i work on this ?

**Comment by Kvaibhav01** [comment]
@rajvijen yes, it will be accepted but if someone provides a correct solution for the same with Java/JavaScript then his/her solution will be added.

@cfsnsalazar @aayushbhan you can start working but make sure you follow the rules and you don't work on a same problem.
